queries/highlights.scm

```diff
@@ -1,132 +1,271 @@
-(comment) @comment
+; Includes
 
 [
-  "("
-  ")"
-  "{"
-  "}"
-  "["
-  "]"
-] @punctuation.bracket
+  "use"
+] @include
 
-[
-  ";"
-  "."
-  ","
-] @punctuation.delimiter
+; Keywords
 
 [
-  "if"
-  "ifdef"
-  "then"
-  "else"
-  "elseif"
-  "end"
-  "try"
-  "while"
-  "for"
-  "use"
-  "as"
-  "var"
-  "let"
-  "embed"
-  "fun"
-  "be"
-  "new"
+  "type"
   "actor"
   "class"
-  "struct"
   "primitive"
   "interface"
   "trait"
-  "type"
+  "struct"
+  "embed"
+  "let"
+  "var"
+  "be"
+  (compile_intrinsic)
+  "as"
+  "consume"
+  "object"
+  "where"
+] @keyword
+
+[
+  "fun"
+] @keyword.function
+
+[
+  "new"
+  "in"
+  "is"
+] @keyword.operator
+
+[
+  "return"
+] @keyword.return
+
+; Qualifiers
+
+[
   "iso"
   "trn"
   "ref"
   "val"
   "box"
   "tag"
+] @type.qualifier
+
+; Conditionals
+
+[
+  "if"
+  "ifdef"
+  "iftype"
+  "then"
+  "else"
+  "elseif"
+  "until"
+  "match"
+  "break"
+] @conditional
+
+(if_statement "end" @conditional)
+
+(iftype_statement "end" @conditional)
+
+(match_expression "end" @conditional)
+
+; Repeats
+
+[
+  "repeat"
+  "while"
+  "for"
+  "continue"
+  "do"
+] @repeat
```

Comment:
we totally forgot, that all the loop constructs can have an `else` clause for the case when there is no iteration done. Will add this as another PR.

Reply:
Ahh..is it after the do block?

Reply:
yeah, exactly like `else` blocks in all the `if` variants.